server/src/main/java/com/cloud/vm/UserVmManagerImpl.java

```diff
@@ -6319,6 +6319,16 @@ private VMInstanceVO preVmStorageMigrationCheck(Long vmId) {
                     + " hypervisors: [%s].", hypervisorType, HYPERVISORS_THAT_CAN_DO_STORAGE_MIGRATION_ON_NON_USER_VMS));
         }
 
+        List<VolumeVO> vols = _volsDao.findByInstance(vm.getId());
+        if (vols.size() > 1) {
+            // OffLineVmwareMigration: data disks are not permitted, here!
+            if (vols.size() > 1 &&
+                    // OffLineVmwareMigration: allow multiple disks for vmware
+                    !(HypervisorType.VMware.equals(hypervisorType) || HypervisorType.KVM.equals(hypervisorType))) {
+                throw new InvalidParameterValueException("Data disks attached to the vm, can not migrate. Need to detach data disks first");
+            }
+        }
+
```

Comment:
@rg9975 will this code block cause any regression on Xenserver/XCP-ng?
Can you please check (comments doesn't seem right), refactor it to a separate method and may add unit tests

Reply:
Looking at the logic and the pre-code, I'm not seeing a reason for this piece of code.  It may have been a change made when trying to fully understand how migration occurred in the Cloudstack framework that was not warranted.  Outside of potentially impacting other hypervisors, its presence doesn't change anything about KVM.  I am removing it in a new commit and if encounter a new issue on retest will identify and re-introduce a more appropriate change.